applications/crossbar/test/cb_harrys_SUITE.erl

```diff
@@ -0,0 +1,187 @@
+  DecodedResp = jiffy:decode(RespBody),
+  Data = get_value(<<"data">>, DecodedResp),
+  AccID = get_value(<<"account_id">>, Data),
+  Token = get_value(<<"auth_token">>, DecodedResp),
+  ok = application:set_env(cb_harrys_tests, account_id, binary_to_list(AccID)),
+  ok = application:set_env(cb_harrys_tests, auth_token, binary_to_list(Token)).
+
+-spec logout_user() ->
+  ok.
+logout_user() ->
+  % TODO: perform api call to logout the user.
+  ok = application:unset_env(cb_harrys_tests, account_id),
+  ok = application:unset_env(cb_harrys_tests, auth_token).
+
+-spec clean_harrys() ->
+  ok.
+clean_harrys() ->
+  {200, Resp} = api_call(get),
+  ExistingHarrys = get_value(<<"data">>, Resp),
+  ok = lists:foreach(
+         fun(Harry) ->
+             {200, _} = api_call(delete, <<>>, get_value(<<"id">>, Harry))
+         end,
+         ExistingHarrys
+        ).
+
+-spec account_id() ->
+  string().
+account_id() ->
+  {ok, AccID} = application:get_env(cb_harrys_tests, account_id),
+  AccID.
+
+-spec auth_token() ->
+  string().
+auth_token() ->
+  {ok, AuthToken} = application:get_env(cb_harrys_tests, auth_token),
+  AuthToken.
+
+-spec build_harry(FName::binary(), LName::binary(), Email::binary()) ->
+  jiffy:json_value().
+build_harry(FName, LName, Email) ->
+  % {"data": {"first_name": "value", "last_name": "value", "email": "value"}}
+  {[{<<"data">>, {[{<<"first_name">>, FName},
+                   {<<"last_name">>, LName},
+                   {<<"email">>, Email}]} }]}.
+
+-spec api_call(Method::method()) ->
+  {integer(), jiffy:jiffy_decode_result()}.
+api_call(Method) ->
+  api_call(Method, <<>>, "").
+
+-spec api_call(Method::method(), ReqBody::binary() | string()) ->
+  {integer(), jiffy:jiffy_decode_result()}.
+api_call(Method, ReqBody) ->
+  api_call(Method, ReqBody, "").
+
+-spec api_call(Method::method(),
+               ReqBody::binary() | string(),
+               HarryID::string() | binary()) ->
+  {integer(), jiffy:jiffy_decode_result()}.
+api_call(Method, ReqBody, HarryID) when is_binary(HarryID) ->
+  api_call(Method, ReqBody, binary_to_list(HarryID));
+api_call(Method, ReqBody, HarryID) ->
+  BaseURL = "http://192.168.99.100:8000/v2/accounts/",
+  URL = lists:flatten([BaseURL, account_id(), "/harrys/", HarryID]),
+  Headers = [{"X-Auth-Token", auth_token()}],
+  Request = case Method of
+              get -> {URL, Headers};
+              _ -> {URL, Headers, "application/json", ReqBody}
+            end,
+  ct:log("Sending ~p request to ~p with:~nHeaders: ~p~nBody: ~p",
+         [Method, URL, Headers, ReqBody]),
+  {ok, {{_Vsn, HTTPCode, _HTTPText}, _RespHeaders, RespBody}} =
+    httpc:request(Method, Request, [], []),
+  DecodedResp = jiffy:decode(RespBody),
+  ct:log("Got ~p response with:~nRespBody: ~p~n", [HTTPCode, DecodedResp]),
+  {HTTPCode, DecodedResp}.
+
+-spec get_value(Key::binary(), Data::jiffy:json_value()) ->
+  term().
+get_value(Key, {Data}) ->
+  proplists:get_value(Key, Data).
```

Comment:
Again, unpacking the internal structure is verboten. I would write this as:

```erlang
get_value(Key, JObj) ->
    kz_json:get_value(Key, JObj).
```

And you'll probably just want to change your `get_value/2` calls to call `kz_json:get_value/2` directly. Also, kz_json has many typed versions of get_value. If you know the value must be a ne_binary, use `get_ne_binary_value`; if an object, `get_json_value`, etc.